src/common/Surface.js

```diff
@@ -139,24 +140,27 @@
             .render()
         ;
 
-        var surfaceButtons = element.selectAll(".surface-button").data(this.buttonAnnotations());
-        surfaceButtons.enter().append("g").classed("surface-button",true)
+        var surfaceButtons = this.buttonContainer.selectAll(".surface-button").data(this.buttonAnnotations());
+        surfaceButtons.enter().append("button").attr("class","surface-button")
             .each(function (button, idx) {
-                context._surfaceButtons[idx] = new Icon()
-                    .faChar(button.char)
-                    .diameter(button.diameter)
-                    .image_colorFill(button.image_colorFill)
-                    .shape_colorFill(button.shape_colorFill)
-                    .shape_colorStroke(button.shape_colorStroke)
-                    .paddingPercent(button.padding)
-                    .shape(button.shape)
-                    .target(this)
-                    .display(context.showTitle())
-                    .render(function(widget) {
-                        widget.element().style("cursor","pointer");
-                        widget.click = function(d) { context.click(d,widget); };
-                    })
-                ;
+                var el = context._surfaceButtons[idx] = d3.select(this)
+                    .attr("class", "surface-button " + (button.class ? button.class : ''))
+                    .attr("id", button.id)
+                    .style('padding', button.padding)
+                    .style('width', button.width)
+                    .style('height', button.height)
+                    .style("cursor","pointer");
+                if (button.font === "FontAwesome") {
```

Comment:
i can prolly make this shorter in both by including some of the EL stuff out side of the if?